### ISSUE TITLE
Grace period not being followed for certain files

### ISSUE THREAD

**Issue by rightaway** [guard/guard-livereload#114]
```
guard 'livereload', grace_period: 5 do
  watch(%r{.+\.(css|html|py)$})
end
```

This is my Guardfile, which reloads the browser anytime a css, html, or py file is changed. It works fine with html or css, waiting the specified 5 seconds before refreshing. But for some reason when a py file is changed, the browser refreshes immediately. The console output reflects that too.

It's a problem because the py file change causes the server to reload, but since the grace period isn't being followed the browser is refreshed before the server has restarted.

**Comment by Jacob-Mango** [comment]
Still not fixed? I am using LinuxMint and it is not working for me. No where else says anything.

**Comment by e2** [comment]
@Jacob-Mango - try a PR with a possible fix: #163 

In your `Gemfile`:

```ruby
gem 'guard-livereload', github: 'guard/guard-livereload', branch: 'e2-guaranteed-sleep'
```

I'm guessing it has to do with the way Guard interrupts the sleep when other files are changed (interactor thread is suspended, resumed).
